packages/mcfunction/src/parser/macro.ts

```diff
@@ -0,0 +1,108 @@
+import * as core from '@spyglassmc/core'
+import { localeQuote, localize } from '@spyglassmc/locales'
+import type { MacroChildNode, MacroKeyNode, MacroNode } from '../node/index.js'
+
+/**
+ * Parse a macro line.
+ */
+export function macro(): core.Parser<MacroNode> {
+	return (
+		src: core.Source,
+		ctx: core.ParserContext,
+	): core.Result<MacroNode> => {
+		const ans: MacroNode = {
+			type: 'mcfunction:macro',
+			range: core.Range.create(src.cursor),
+			children: [],
+		}
+
+		// Skip the starting '$'
+		ans.children.push({
+			type: 'mcfunction:macro_child/macro',
+			range: core.Range.create(src.cursor, src.cursor + 1),
+			value: '$',
+		})
```

Comment:
Reusing the same AST node as macro keys is somehow even more of a hack. I suggest either adding the old node type back or colorizing the `$` without there being a node for it.